### DISCUSSION TITLE
Exclude code files that generate error messages?

### DISCUSSION THREAD

**Discussion by JustineSondore** [stryker-mutator/stryker-net#2017] [Q&A] [answered]
I'm using stryker.net 1.5.2. on an .Net Core solution that includes several projects.

On one of the projects-under-test stryker generates errors:
[14:09:30 WRN] Stryker.NET encountered an compile error in C:\SomePath\ObscuredProjectName\Folder\Folder1\Folder2\CodeFile.cs (at 34:25) with message: Use of unassigned local variable 'someVariable' (Source code: someVariable)
[14:09:30 WRN] Safe Mode! Stryker will try to continue by rolling back all mutations in method. This should not happen, please report this as an issue on github with the previous error message.

I tried to exclude the file by using a flag:
dotnet-stryker -p "ObscuredProjectName.csproj" -m "!Folder/Foler1/Folder2/CodeFile.cs"
but still errors are generated.

I did a little test on another project that stryker runs fine on without errors and excluded a file in a similar manner; the result from that project now had an expected N/A for the file; but also "Ignored" value was a number, not zero; does this imply mutations are still generated, just that tests are not run against them?

So the question is, if I'm gathering correctly, that using -m is not a way to get around above mentioned kind of errors?

Am I missing another way to exclude some parts of code from begin mutated?

**Comment by dupdob** [comment, marked as answer]
The short answer is that, yes, you understand this correctly: mutations are still generated but ignored.

As of now, there is no way to completely exclude file (or part of a file) from being mutated. One can only request Stryker to ignore some mutations (see [documentation](https://stryker-mutator.io/docs/stryker-net/ignore-mutations)).
This is on purpose as it still tracks generated mutations, allowing one to decide to stop ignoring them, and we estimate there is no benefit in actually exclude files from being mutated.

Regarding the _error_ you are reporting, I would like to stress out this is a _warning_ because Stryker faced an error for which it could not identify which mutation was the cause. Therefore, it needs to removes all mutation from the concerned method.
We ask people to report this kind of errors in order to help us improve the efficiency of the mutation process.
